aspnet-core/src/TalentV2.Core/WebServices/ExternalServices/Autobot/AutobotService.cs

```diff
@@ -100,6 +123,194 @@ public async Task<T> ExtractCVInformationAsync<T>(Stream stream, string fileName
+
+
+        public async Task<CVScanResultFromFireBase> ExtractCVFromFirebaseAsync(string fileUrl)
+        {
+            var fileName = Path.GetFileName(new Uri(fileUrl).AbsolutePath);
+            var fileNameDecoded = WebUtility.UrlDecode(fileName);
+            var fileBytes = await HttpClient.GetByteArrayAsync(fileUrl);
+
+
+            var requestUrl = $"{HttpClient.BaseAddress}{ExtractCV}";
+            const int maxRetries = 5;
+            int delayTime = 10000; // 10s
+            int attempt = 0;
+            bool isComplete = false;
+            while (attempt < maxRetries && !isComplete)
+            {
+                try
+                {
+                    using (var content = new MultipartFormDataContent())
+                    {
+                        content.Add(new ByteArrayContent(fileBytes), "file", fileNameDecoded);
+                        var response = await HttpClient.PostAsync(requestUrl, content);
+                        if (response.StatusCode == HttpStatusCode.TooManyRequests)
+                        {
+                            attempt++;
+                            if (attempt == maxRetries)
+                            {
+                                logger.LogError($"Attempt {attempt} is maximum number of retries");
+                                break;
+                            }
+
+                            logger.LogError($"Attempt {attempt} failed due to AI request-limiting. Retrying in {delayTime / 1000} seconds...");
+                            delayTime = Math.Min(delayTime * 2, 60000);
+                            await Task.Delay(delayTime);
+                        }
+                        else
+                        {
+                            response.EnsureSuccessStatusCode();
+                            var jsonResponse = await response.Content.ReadAsStringAsync();
+                            var result = JsonConvert.DeserializeObject<CVScanResultFromFireBase>(jsonResponse);
+                            isComplete = true;
+                            return result;
+                        }
+                    };
+
+                }
+
+                catch (TaskCanceledException e)
+                {
+
+                    logger.LogError("TaskCanceledException occurred: {Exception}", e);
+                }
+                catch (HttpRequestException ex)
+                {
+                    logger.LogError("HttpRequestException occurred: {Exception}", ex);
+
+                    if (ex.InnerException is SocketException socketEx)
+                    {
+                        logger.LogError($"SocketException: {socketEx.Message}");
+                    }
+                }
+                catch (Exception ex)
+                {
+                    logger.LogError("Exception occurred: {Exception}", ex);
+
+                }
+            }
+
+            logger.LogError($"Failed to extract CV from Firebase at {fileName}");
+            return null;
+        }
```

Comment:
setup retry policy cho việc call api thay vì sử dụng while loop để xử lý failures
refer: https://www.codeproject.com/Articles/5378791/How-to-Use-Polly-In-Csharp-Easily-Handle-Faults-An

không cần xử lý cho từng loại exception, chỉ cần ghi log cho tất cả các loại exception để khi check log dễ tìm